src/mesh/interpolation.cxx

```diff
@@ -222,8 +228,134 @@ const Field3D interp_to(const Field3D &var, CELL_LOC loc)
   return var;
 }
 
-const Field2D interp_to(const Field2D &var, CELL_LOC UNUSED(loc)) {
-  // Currently do nothing
+const Field2D interp_to(const Field2D &var, CELL_LOC loc, REGION region)
+{
+  // Note Field2D has no z-variation, so nothing needs to be done for
+  // interpolation in the z-direction
+  if(mesh->StaggerGrids && loc != CELL_ZLOW && (var.getLocation() != loc)) {
+
+    // Staggered grids enabled, and need to perform interpolation
+    TRACE("Interpolating %s -> %s", strLocation(var.getLocation()), strLocation(loc));
+
+    Field2D result(var.getMesh());
+
+    if (region != RGN_NOBNDRY) {
+      // result is requested in some boundary region(s)
+      result = var; // NOTE: This is just for boundaries. FIX!
```

Comment:
Is now a good time to address the "fix" mentioned in the comments?

Reply:
Is there an obvious fix? I think the only correct thing to do is to apply a boundary condition after interpolating. I don't think the input Field3D knows its boundary conditions at the moment? If it did maybe we could try to do something clever here, but otherwise I think we just have to define the guard cells of the result of interpolation as invalid, and make the user apply a boundary condition if they need one.